src/main/java/com/basho/riak/client/api/commands/kv/MultiCommand.java

```diff
@@ -0,0 +1,408 @@
+package com.basho.riak.client.api.commands.kv;
+
+import com.basho.riak.client.api.RiakCommand;
+import com.basho.riak.client.api.commands.ListenableFuture;
+import com.basho.riak.client.api.commands.RiakOption;
+import com.basho.riak.client.core.RiakCluster;
+import com.basho.riak.client.core.RiakFuture;
+import com.basho.riak.client.core.RiakFutureListener;
+import com.basho.riak.client.core.query.Location;
+
+import java.util.*;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * @param <BaseCommand> The type of the individual command you are trying to repeat
+ * @param <BaseBuilder> The type of the builder for an individual {@link BaseCommand}
+ * @param <ResponseType> The return type of the grouped "multi" command
+ * @param <BaseResponseType> The return type of an individual {@link BaseCommand}
+ */
+/*
+
+ */
+abstract class MultiCommand<BaseCommand extends RiakCommand<BaseResponseType, Location>,
+                                   BaseBuilder extends KvBuilderBase<BaseCommand>,
+                                   ResponseType extends Iterable<RiakFuture<BaseResponseType, Location>>,
+                                   BaseResponseType>
+        extends RiakCommand<ResponseType, List<Location>>
+{
+    public static final int DEFAULT_MAX_IN_FLIGHT = 10;
+
+    protected final ArrayList<Location> locations;
+    protected final Map<RiakOption<?>, Object> options = new HashMap<>();
+    protected final int maxInFlight;
+
+    @SuppressWarnings("unchecked")
+    protected MultiCommand(Builder builder)
+    {
+        this.locations = builder.locations;
+        this.options.putAll(builder.options);
+        this.maxInFlight = builder.maxInFlight;
+    }
+
+
+    @Override
+    protected RiakFuture<ResponseType, List<Location>> executeAsync(final RiakCluster cluster)
+    {
+        List<BaseCommand> operations = buildOperations();
+        MultiFuture future = new MultiFuture(locations);
+
+        Submitter submitter = new Submitter(operations, maxInFlight, cluster, future);
+
+        Thread t = new Thread(submitter);
+        t.setDaemon(true);
```

Comment:
I modified the Python client to *not* use daemon threads, since they could hide programming errors (not closing the client correctly, for instance). Should `setDaemon(false)` be used here?